daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala

```diff
@@ -588,15 +594,14 @@ class ContractStateMachineSpec extends AnyWordSpec with Matchers with TableDrive
     for {
       next <- node match {
         case create: Node.Create =>
-          state.visitCreate(create.templateId, create.coid, create.key)
-        case exercise: Node.Exercise => state.handleExercise((), exercise)
-        case fetch: Node.Fetch => state.handleFetch(fetch)
+          state.handleCreate(create)
+        case exercise: Node.Exercise =>
+          state.handleExercise((), exercise)
+        case fetch: Node.Fetch =>
+          state.handleFetch(fetch)
         case lookup: Node.LookupByKey =>
           val gkey = GlobalKey.assertBuild(lookup.templateId, lookup.key.key)
-          if (state.mode == ContractKeyUniquenessMode.Strict)
-            state.handleLookup(lookup)
-          else
-            state.handleLookupWith(lookup, resolver(gkey))
+          state.handleLookup(lookup)
```

Comment:
Are you sure that we don't need this distinction here? I thought that it's necessary in ContractKeyUniquenessMode.Off.